components/ai_chat/resources/page/components/edit_input/index.tsx

```diff
@@ -19,6 +19,21 @@ interface Props {
 
 function EditInput(props: Props) {
   const [text, setText] = React.useState(props.text)
+  const textareaRef = React.useRef<HTMLTextAreaElement>(null)
+
+  React.useEffect(() => {
+    if (textareaRef.current) {
+      /**
+       * When editing, we want the cursor to be positioned
+       * at the end of the text. Typically we would also call
+       * focus() here, but that isn't necessary in this case
+       * because the textarea already has [autoFocus] set.
+       */
+      const length = textareaRef.current.value.length
+      textareaRef.current.setSelectionRange(length, length)
+    }
+  }, [])
```

Comment:
Instead of a Ref just to read its value once in a useEffect, I'm wondering if this will work just as well with a `const textAreaRef = React.useCallback(textAreaRef => { ...`?

Reply:
Good call!